products/compute/api.yaml

```diff
@@ -7553,6 +7553,167 @@ objects:
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          Name of the resource; provided by the client when the resource is
+          created. The name must be 1-63 characters long, and comply with
+          RFC1035. Specifically, the name must be 1-63 characters long and match
+          the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?` which means the
+          first character must be a lowercase letter, and all following
+          characters must be a dash, lowercase letter, or digit, except the last
+          character, which cannot be a dash.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: |
+          An optional description of this resource. Provide this property when
+          you create the resource.
+      - !ruby/object:Api::Type::Enum
+        name: 'networkEndpointType'
+        required: true
+        description: |
+          Type of network endpoints in this network endpoint group.
+        values:
+          - :INTERNET_IP_PORT
+          - :INTERNET_FQDN_PORT
+          - :SERVERLESS
+          - :NON_GCP_PRIVATE_IP_PORT
+          - :GCE_VM_IP_PORT
+      - !ruby/object:Api::Type::NestedObject
+        name: 'cloudRun'
+        exactly_one_of:
+          - cloudRun
+          - cloudFunction
+          - appEngine
+        description: |
+          Only valid when networkEndpointType is "SERVERLESS".
+          Only one of cloud_run, app_engine or cloud_function may be set.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'service'
+            description: |
+              Cloud Run service is the main resource of Cloud Run.
+              The service must be 1-63 characters long, and comply with RFC1035.
+              Example value: "run-service".
+          - !ruby/object:Api::Type::String
+            name: 'tag'
+            description: |
+              Cloud Run tag represents the "named-revision" to provide
+              additional fine-grained traffic routing information.
+              The tag must be 1-63 characters long, and comply with RFC1035.
+              Example value: "revision-0010".
+          - !ruby/object:Api::Type::String
+            name: 'urlMask'
+            description: |
+              A template to parse service and tag fields from a request URL. 
+              URL mask allows for routing to multiple Run services without having 
+              to create multiple network endpoint groups and backend services.
+
+              For example, request URLs "foo1.domain.com/bar1" and "foo1.domain.com/bar2" 
+              an be backed by the same Serverless Network Endpoint Group (NEG) with 
+              URL mask ".domain.com/". The URL mask will parse them to { service="bar1", tag="foo1" } 
+              and { service="bar2", tag="foo2" } respectively. 
+      - !ruby/object:Api::Type::NestedObject
+        name: 'appEngine'
```

Comment:
Right now, you don't have any required fields (or an at_least_one_of) inside of this block. That implies that sending an empty `appEngine {}` block to the API is possible and has a specific meaning. Is that true? If so, can you please add details in a comment?

Reply:
i though if it was empty terraform drops that object from the API call, i haven't defined a default for it